plasma_framework/contracts/poc/fast_exits/Liquidity.sol

```diff
@@ -0,0 +1,186 @@
+     * @param utxoPosInput position of the output that created the inputs for second transaction
+    */
+    function verifyOwnership(
+        bytes memory rlpInputCreationTx,
+        uint256 utxoPosInput
+    ) internal {
+
+        PaymentTransactionModel.Transaction memory decodedFirstTx
+        = PaymentTransactionModel.decode(rlpInputCreationTx);
+        PosLib.Position memory position = PosLib.decode(utxoPosInput);
+        uint16 firstTransactionOutputIndex = position.outputIndex;
+
+        FungibleTokenOutputModel.Output memory outputFromFirstTransaction
+        = decodedFirstTx.outputs[firstTransactionOutputIndex];
+        address ownerFirstTxOutput = PaymentTransactionModel.getOutputOwner(outputFromFirstTransaction);
+        require(
+            ownerFirstTxOutput == msg.sender,
+            "Was not called by the first Tx owner"
+        );
+    }
+
+    /**
+     * @notice Verify the First Tx provided is valid
+     * @param utxoPosInput position of the output that created the inputs for second transaction
+     * @param rlpInputCreationTx RLP-encoded first transaction that transfers to this contract
+     * @param inputCreationTxInclusionProof First transactions inclusion proofs
+    */
+    function verifyTxValidity(
+        uint256 utxoPosInput,
+        bytes memory rlpInputCreationTx,
+        bytes memory inputCreationTxInclusionProof
+    ) internal {
+        PosLib.Position memory utxoDecoded = PosLib.decode(utxoPosInput);
+        utxoDecoded.outputIndex = 0;
+        (bytes32 root, ) = plasmaFrameworkInstance.blocks(utxoDecoded.blockNum);
+        require(root != bytes32(""), "Failed to get root of the block");
+        bool txExists = Merkle.checkMembership(
+            rlpInputCreationTx,
+            utxoDecoded.txIndex,
+            root,
+            inputCreationTxInclusionProof
+        );
+        require(
+            txExists,
+            "Provided Transaction isn't finalized or doesn't exist"
+        );
+    }
+
+    /**
+     * @notice func that calls omg-contracts to start the exit
+     * @param utxoPosToExit position of the output which the contract has to exit
+     * @param rlpOutputTxToContract RLP-encoded transaction that creates the output for the contract
+     * @param outputTxToContractInclusionProof Second Transaction's inclusion proof
+    */
+    function runExit(
+        uint256 utxoPosToExit,
+        bytes memory rlpOutputTxToContract,
+        bytes memory outputTxToContractInclusionProof
+    ) private {
+        PaymentStandardExitRouterArgs.StartStandardExitArgs memory s;
+        s.utxoPos = utxoPosToExit;
+        s.rlpOutputTx = rlpOutputTxToContract;
+        s.outputTxInclusionProof = outputTxToContractInclusionProof;
+        paymentExitGame.startStandardExit.value(msg.value)(s);
+    }
+
+    function getContractBalance() public view returns(uint256) {
+        return address(this).balance;
+    }
```

Comment:
I believe that we do not need this helper

Reply:
true. removed `8d760f9`